public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchLogsQueryRunner.ts

```diff
@@ -168,16 +114,17 @@ export class CloudWatchLogsQueryRunner extends CloudWatchRequest {
       return Date.now() >= startTime.valueOf() + rangeUtil.intervalToMs(this.logsTimeout);
     };
 
+    // run with retry will retry any failed start queries due to rate limiting
     return runWithRetry(
-      (targets) => {
-        return this.makeLogActionRequest('StartQuery', targets, queryFn, options);
-      },
+      (targets) => this.makeLogActionRequest('StartQuery', targets, queryFn, options),
       startQueryRequests,
       timeoutFunc
     ).pipe(
-      mergeMap(({ frames, error }: { frames: DataFrame[]; error?: DataQueryError }) =>
-        this.getQueryResults({ frames, logQueries, timeoutFunc, error, queryFn })
-      ),
+      // once we've started the query, we need to poll for the results
+      mergeMap((startQueryResponse) => {
+        return this.getQueryResults({ logQueries, timeoutFunc, queryFn, startQueryResponse })
```

Comment:
the main change here is that we now pass the whole response body from StartQuery, so that if there is an error in the response we'll be able to show it even as we poll for more responses that may not have errored.